src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -87,6 +87,54 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (path == null)
+                throw new ArgumentException(nameof(path));
+
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (basePath.Contains('\0') || path.Contains('\0'))
+                throw new ArgumentException(SR.Argument_InvalidPathChars);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                // paths starting with \\?\ & \\.\
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
+            }
+            else if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])))
+            {
+                // path is current drive rooted i.e. starts with \
+                return RemoveRelativeSegments(CombineNoChecks(GetPathRoot(basePath), path.Substring(1)));
+            }
+            else if (length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar)
+            {
+                Debug.Assert(length == 2 || !PathInternal.IsDirectorySeparator(path[2]));
+                
+                // same and different specific drive rooted path
+                if (GetPathRoot(path.AsReadOnlySpan()) == GetPathRoot(basePath.AsReadOnlySpan()))
+                {
+                    // paths such as C:foo
+                    return RemoveRelativeSegments(CombineNoChecks(basePath, path.Substring(2)), basePath.Length);
```

Comment:
You can use ReadOnlySpan and slice here. Substring() will unnecessarily allocate.

Reply:
Additionally, as above, use GetFullPath on the result, not RemoveRelativeSegments.

Reply:
This case is `C:Foo` and `C:\Bar`, as they DO match, you're returning `C:\Bar\Foo`.

Reply:
isn't the correct answer ```C:\Bar\Foo```